### ISSUE TITLE
The aws_ecs_service doesn't wait for a new computed aws_ecs_task_definition

### ISSUE THREAD

**Issue by killercentury** [hashicorp/terraform#2694]
I found that the aws_ecs_service doesn't wait for the new computed aws_ecs_task_definition. It just uses the old aws_ecs_task_definition at the first time I run plan and apply. When I run plan and apply again, it will pick up the new aws_ecs_task_definition.

Here is my sample config:
```
resource "aws_ecs_task_definition" "terraform_ecs_demo" {
  family = "terraform-ecs-demo"
  container_definitions = "${file("ecs-container-definitions.json")}"
}

resource "aws_ecs_service" "terraform_ecs_demo" {
  name = "terraform-ecs-demo"
  cluster = "${aws_ecs_cluster.default.id}"
  task_definition = "${aws_ecs_task_definition.terraform_ecs_demo.arn}"
  desired_count = 1
}
```

I also tried adding `depends_on = ["aws_ecs_task_definition.terraform_ecs_demo"]` to the service, but same result.

**Comment by roosmaa** [comment]
I'm also getting this. Could this be that terraform `aws_ecs_task_definition` doesn't know that when updating the definition it should also bump the revision number in it's arn, thus `aws_ecs_service` can't pick up the change?

**Comment by CpuID** [comment]
yep, noticing the same here, 0.6.3. have been manually updating the task def on the service so far, but if another apply does the job makes sense.

**Comment by dhumphreys** [comment]
I am also experiencing this. Any updates?

**Comment by brikis98** [comment]
I'm having the same issue here on 0.6.4. I update the `aws_ecs_task_definition`, run `terraform apply`, and the new revision shows up in ECS, but the service is not updated. If I run `terraform apply` again, the service is updated, and the new revision gets deployed. I'll poke around the code to see if I can spot the bug and submit a PR, but I'm new to Go, so pointers are welcome.

**Comment by catsby** [comment]
When you update the task definition and `plan`, does a change show up in the `aws_ecs_service`? 

I would suspect not. The task definition's ARN isn't changing, is it? Does someone have a valid `container_definitions` file I can use?

**Comment by CpuID** [comment]
@catsby no, not in this case. plan and apply yield the same result most of the time (just a task def change, without a service bump for the task version)

example of an aws_ecs_task_definition (including an inline container_definitions entry):

```
resource "aws_ecs_task_definition" "nginx" {
  family = "nginx"
  container_definitions = <<DEF
[
  {
    "name": "nginx",
    "image": "nginx:1.9.0",
    "cpu": 60, 
    "memory": 512,
    "essential": true,
    "environment": [
      {   
        "name": "SERVICE_NAME",
        "value": "nginx"
      }
    ],  
    "portMappings": [
      {   
        "containerPort": 80,
        "hostPort": 80
      }   
    ]   
  }
]
DEF
}
```

**Comment by listenrightmeow** [comment]
:+1:

**Comment by sunilvigneshr** [comment]
Experiencing this. Any Updates?

**Comment by buzzedword** [comment]
Also seeing this. :+1:

**Comment by radeksimko** [comment]
Reproduced, I think that marking all fields in `aws_ecs_task_definition` as `ForceNew: true` would probably fix this.

After all, we're effectively creating a new resource with a new ID anyway, so it won't make much difference IMO.
I will have a look into it.

**Comment by radeksimko** [comment]
btw. I think the reason why `aws_ecs_service` doesn't get updated in the same run is because it has no idea whether ARN of the referenced `aws_ecs_task_definition` will be re-computed during `plan`. It is afaik only re-computed when the whole resource gets re-created.

When it actually happens - during `apply`, it's too late and terraform is just executing expected plan which won't change at that point.

@catsby Maybe there should be a way to trigger "recomputation" of `Computed` fields under certain conditions, what do you think?

**Comment by radeksimko** [comment]
See #3924 which is fixing this (verified by a simple acceptance test included in the PR).

**Comment by jbergknoff** [comment]
Should that fix be in 0.6.8? I just upgraded but still experience the issue: the task definition updates but the service depending on that task definition does not update. I need to `apply` a second time to make the service update.

**Comment by radeksimko** [comment]
@jbergknoff This should be fixed in `0.6.8`. I am personally using both `ecs_service` & `ecs_task_definition` and it works fine for me.

If you're still experiencing any issues, report it as a new issue please. Ideally include the terraform code used (minus any secrets), so we can reproduce it and fix it.
